internal/server/models/providergroup.go

```diff
@@ -0,0 +1,19 @@
+package models
+
+import (
+	"github.com/infrahq/infra/uid"
+)
+
+// ProviderGroup is a local copy of the of a group from an identity provider
+// See this diagram for more details about how this model relates to a group
+// https://github.com/infrahq/infra/blob/main/docs/dev/identity-provider-tracking.md
+type ProviderGroup struct {
+	Model
+	OrganizationMember
```

Comment:
I think we don't need `OrganizationMember` here, similar to `ProviderUser`  and `identities_groups`. Since this belongs to a provider, the organization_id will always be the one associated with the provider_id.

Do we need the whole `Model` ? I see we are using the `ID` from model right now, but I suspect we don't need the other fields, similar to `ProviderUser`, and `identities_groups`.

I wonder if we replace the `ID` with something like `user_id` and `group_id` if we could remove some of the sub-selects and outer joins.

Reply:
We will want the `Model` for SCIM groups, we need a way to track when provider group was created, updated, and disabled to keep it in sync with the external group when SCIM requests come in.

The `OrganizationMember` is on the provider group because it is possible to `ListProviderGroups` by name and the provider ID isn't required. This was a defense against that case. I'll just remove the ability to list by name for now instead.